src/App.tsx

```diff
@@ -1,11 +1,51 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+import * as todoService from './api/todos';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoNotification } from './components/TodoNotification';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [filterField, setFilterField] = useState('all');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  function loadTodos() {
+    setLoading(true);
+
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .finally(() => setLoading(false));
+
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }
+
+  const filterTodos = (value: string) => {
+    switch (value) {
+      case 'active':
+        return [...todos].filter(todo => !todo.completed);
```

Comment:
```suggestion
        return todos.filter(todo => !todo.completed);
```
it's redundant, filter method doesn't mutate the array